internal/lock.go

```diff
@@ -26,24 +26,24 @@ type config struct {
 	Resource []Resource
 }
 
-func NewLock(path string, newOk bool) (*Lock, error) {
-	_, error := os.Stat(path)
-	if os.IsNotExist(error) {
-		if newOk {
-			return &Lock{path: path}, nil
-		} else {
-			return nil, fmt.Errorf("file '%s' does not exist", path)
+func NewLock(path string, create bool) (*Lock, error) {
+	if !create {
+		_, err := os.Stat(path)
+		if err != nil {
+			return nil, err
 		}
 	}
+
 	var conf config
-	file, err := os.Open(path)
-	if err != nil {
-		return nil, error
-	}
-	d := toml.NewDecoder(file)
-	err = d.Decode(&conf)
-	if err != nil {
-		return nil, err
+	if !create {
+		data, err := os.ReadFile(path)
+		if err != nil {
+			return nil, fmt.Errorf("cannot open file '%s': %s", path, err)
+		}
+		err = toml.Unmarshal(data, &conf)
+		if err != nil {
+			return nil, fmt.Errorf("invalid lock file '%s': %s", path, err)
+		}
 	}
 
 	return &Lock{path: path, conf: conf}, nil
```

Comment:
This whole function now does nothing if `!create` (the main logic of this function is split into 2 blocks, both of which are protected by `if !create`), which seems odd.

Is there a reason to call this function if you don't want to create a lock?

Additionally, it seems like you changed the meaning of the second parameter from `newOk` (meaning: if `!newOk` and we have to create a file, return an error) to `create` (meaning: if `!create`, never return an error).  Can you explain why this is needed?